test/unittests/test_analyse_xml_data/test_extract_links_from_XML.py

```diff
@@ -102,28 +103,54 @@ def test_extract_ids_from_text_prop_with_several_text_links() -> None:
         'class="salsah-link" href="IRI:res_B_18:IRI">res_B_18</a></text></text-prop>'
     )
-    res = _extract_ids_from_text_prop(test_ele)
-    assert unordered(res) == [{"res_A_18"}, {"res_B_18"}]
+    res = _create_class_instance_text_prop("res_C_18", test_ele)
+    res_ids = [x.object_link_ids for x in res]
+    assert unordered(res_ids) == [{"res_A_18"}, {"res_B_18"}]
 
 
-def test_extract_one_id_resptr_prop() -> None:
+def test_create_class_instance_resptr_link_one_link() -> None:
     test_ele = etree.fromstring(
         '<resptr-prop xmlns="https://dasch.swiss/schema" '
         'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name=":hasResource1"><resptr '
         'permissions="prop-default">res_C_15</resptr></resptr-prop>'
     )
-    res = _extract_ids_from_one_resptr_prop(test_ele)
-    assert unordered(res) == ["res_C_15"]
+    res = _create_class_instance_resptr_link("res_A_15", test_ele)
+    assert res[0].object_id == "res_C_15"
 
 
-def test_extract_several_id_resptr_prop() -> None:
+def test_create_class_instance_resptr_link_several() -> None:
     test_ele = etree.fromstring(
         '<resptr-prop xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
         'name=":hasResource1"><resptr permissions="prop-default">res_A_13</resptr><resptr '
         'permissions="prop-default">res_B_13</resptr><resptr permissions="prop-default">res_C_13</resptr></resptr-prop>'
     )
-    res = _extract_ids_from_one_resptr_prop(test_ele)
-    assert unordered(res) == ["res_A_13", "res_B_13", "res_C_13"]
+    res = _create_class_instance_resptr_link("res_D_13", test_ele)
+    assert res[0].object_id == "res_A_13"
+    assert res[1].object_id == "res_B_13"
+    assert res[2].object_id == "res_C_13"
+
+
+def test_create_info_from_xml_for_graph_check_UUID_in_root() -> None:
+    root = etree.fromstring(
+        b'<?xml version="1.0" encoding="UTF-8"?><knora xmlns="https://dasch.swiss/schema" '
+        b'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://dasch.swiss/schema '
+        b'https://raw.githubusercontent.com/dasch-swiss/dsp-tools/main/src/dsp_tools/resources/schema/data.xsd" '
+        b'shortcode="0700" default-ontology="simcir"><resource label="res_A_11" restype=":TestThing" id="res_A_11" '
+        b'permissions="res-default"><resptr-prop name=":hasResource1"><resptr '
+        b'permissions="prop-default">res_B_11</resptr></resptr-prop></resource><resource label="res_B_11" '
+        b'restype=":TestThing" id="res_B_11" permissions="res-default"><text-prop name=":hasRichtext"><text '
+        b'permissions="prop-default" encoding="xml">Start text<a class="salsah-link" '
+        b'href="IRI:res_C_11:IRI">res_C_11</a>end text.</text></text-prop></resource><resource label="res_C_11" '
+        b'restype=":TestThing" id="res_C_11" permissions="res-default"></resource></knora>'
```

Comment:
any reason you don't use multiline (triple-quote) strings?

Reply:
In triplequote string the new lines are actually encoded newlines. In this case I would prefer not to encode any tabs or linebreaks, so I prefer a single-line string

Reply:
True.